src/solver/where/backward_where.cpp

```diff
@@ -0,0 +1,245 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "miopen/kernel_info.hpp"
+#include "miopen/mlo_internal.hpp"
+#include "miopen/tensor.hpp"
+#include "miopen/tensor_view_utils.hpp"
+#include "miopen/where/problem_description.hpp"
+#include <cstddef>
+#include <miopen/datatype.hpp>
+#include <miopen/kernel_build_params.hpp>
+#include <miopen/where/invoke_params.hpp>
+#include <miopen/where/solvers.hpp>
+#include <miopen/where.hpp>
+#include <miopen/target_properties.hpp>
+#include "../kernels/tensor_view.hpp"
+
+#define LOCAL_SIZE 256
+
+namespace miopen {
+
+namespace solver {
+
+namespace where {
+
+bool WhereBackward::IsApplicable(const ExecutionContext& context,
+                                 const miopen::where::BackwardProblemDescription& problem) const
+{
+    auto ignore = context;
+
+    bool is_all_contiguous = miopen::where::isAllContiguous(problem.GetInputGrad_tv(),
+                                                            problem.GetOtherGrad_tv(),
+                                                            problem.GetCondition_tv(),
+                                                            problem.GetOutputGrad_tv());
+    bool is_all_broadcasted_contiguous =
+        miopen::where::isAllBroadcastedContiguous(problem.GetInputGrad_tv(),
+                                                  problem.GetOtherGrad_tv(),
+                                                  problem.GetCondition_tv(),
+                                                  problem.GetOutputGrad_tv());
+
+    if(!is_all_broadcasted_contiguous && !is_all_contiguous)
+        return false;
+    if(!problem.IsSameType())
+        return false;
+    if(!problem.IsAllPacked())
+        return false;
+    return true;
+}
+
+ConvSolution
+WhereBackward::GetSolution(const ExecutionContext& context,
+                           const miopen::where::BackwardProblemDescription& problem) const
+{
+    std::ignore = context;
+
+    auto cond_contig_size  = miopen::where::checkBroadcastedContiguous(problem.GetCondition_tv());
+    bool is_all_contiguous = miopen::where::isAllContiguous(problem.GetInputGrad_tv(),
+                                                            problem.GetOtherGrad_tv(),
+                                                            problem.GetCondition_tv(),
+                                                            problem.GetOutputGrad_tv());
+    bool is_all_broadcasted_contiguous =
+        miopen::where::isAllBroadcastedContiguous(problem.GetInputGrad_tv(),
+                                                  problem.GetOtherGrad_tv(),
+                                                  problem.GetCondition_tv(),
+                                                  problem.GetOutputGrad_tv());
+    bool is_condition_broadcasted = miopen::where::isConditionBroadcasted(
```

Comment:
If you change the following functions to member functions of ProblemDescription, 
you can use them as follows.
```
    auto cond_contig_size              = problem.checkBroadcastedContiguous();
    bool is_all_contiguous             = problem.isAllContiguous();
    bool is_all_broadcasted_contiguous = problem.isAllBroadcastedContiguous();
    bool is_condition_broadcasted      = problem.isConditionBroadcasted();
```
If you don't think it's suitable, you can ignore my opinion.

Reply:
changed as per your comments